README.md

```diff
@@ -0,0 +1,406 @@
+# PN532 NFC Library
+
+This library allows for integration of NXP's [PN532/C106 NFC Module](http://www.nxp.com/documents/short_data_sheet/PN532_SDS.pdf) in an Electric Imp project.  This device supports a wide range of NFC technologies, including read, write, and emulation support for MIFARE, FeliCa, and ISO/IEC 14443A tags.  It also supports peer-to-peer communication and has power-saving options.
+
+This library currently contains two classes:
+
+- The [`PN532`](#pn532-class) class contains the base code needed to initialize the PN532, communicate with it, read from generic NFC tags, and enable power-saving features.
+- The [`PN532MifareClassic`](#pn532-mifare-classic-class) class contains functions needed to read and write to [MIFARE Classic](http://www.nxp.com/products/identification_and_security/smart_card_ics/mifare_smart_card_ics/mifare_classic/) tags, which support read/write storage.  It also serves as an example of how to build on the `PN532` class to interface with the many other protocols and features that the PN532 supports.
+
+### Examples
+
+For an example of using the `PN532MifareClassic` class to build an ID card reader, see the [MIFARE reader example](examples/MIFARE).
+
+For an example extension class supporting PN532 card emulation mode, see the [`PN532CardEmulator`](examples/cardEmulator) class.  It serves as an example of how to build on the `PN532` class to interface with the many other protocols and features that the PN532 supports.
+
+# PN532 Class
+
+## Constructor: PN532(*spi, ncs, rstpd, irq, callback*)
+
+Creates and initializes an object representing the PN532 NFC device.
+
+### Parameters
+
+- *spi*: A [SPI object](https://electricimp.com/docs/api/hardware/spi/) pre-configured with the flags `LSB_FIRST | CLOCK_IDLE_HIGH` and a clock rate.  The PN532 supports clock rates up to 5 MHz.
+- *ncs*: A pin connected to the PN532's not-chip-select line.
+- *rstpd*: A pin connected to the PN532's RSTPDN (Reset/Power-Down) pin.  This can be null if this the RSTPDN pin will not be under software control.
```

Comment:
It's very important to have a consistent convention for naming electrical nets and pins that are active-low. Chip Select, for example, is active-low, so you've named it "ncs" for "negative chip select". Other conventions are "cs_l" (my favorite), or "cs#" (gross). In the case of the chip select, you're mimicking the datasheet, which is good. 

It looks like RSTPD is also active-low, so you should name all references to it with something that shows this: "nRSTPD" or "rstpd_l".